### DISCUSSION TITLE
Migrating from CRA to Next.js 13 with App Router

### DISCUSSION THREAD

**Discussion by devevignesh** [vercel/next.js#52193] [Help]
### Summary

We currently have a React app that was built on top of CRA, and our Django server serves the React build artifacts as static files. However, we are now planning to migrate from CRA to the Next.js 13 app router.

In React, running yarn build creates an index.html file in the src/build folder. However, Next.js creates HTML files for each route (including index.html, dashboard.html, and profile.html) when using [static exports](https://nextjs.org/docs/app/building-your-application/deploying/static-exports).

Our Django app structure is as follows:

├── app/  
│   └── templates/  
│       └── index.html (created with yarn build in React)  
└── static/  
    ├── css  
    ├── js  
    └── img  
 
I would appreciate your advice on how to proceed with this use case. Would there be any disadvantages if I choose to use "static exports"?

**Comment by devevignesh** [comment]
@leerob could you please help me on this?

**Comment by masterbater** [comment]
> ### Summary
> We currently have a React app that was built on top of CRA, and our Django server serves the React build artifacts as static files. However, we are now planning to migrate from CRA to the Next.js 13 app router.
> 
> In React, running yarn build creates an index.html file in the src/build folder. However, Next.js creates HTML files for each route (including index.html, dashboard.html, and profile.html) when using [static exports](https://nextjs.org/docs/app/building-your-application/deploying/static-exports).
> 
> Our Django app structure is as follows:
> 
> ├── app/ │ └── templates/ │ └── index.html (created with yarn build in React) └── static/ ├── css ├── js └── img
> 
> I would appreciate your advice on how to proceed with this use case. Would there be any disadvantages if I choose to use "static exports"?
> 
> ### Additional information
> _No response_
> 
> ### Example
> _No response_



> ### Summary
> We currently have a React app that was built on top of CRA, and our Django server serves the React build artifacts as static files. However, we are now planning to migrate from CRA to the Next.js 13 app router.
> 
> In React, running yarn build creates an index.html file in the src/build folder. However, Next.js creates HTML files for each route (including index.html, dashboard.html, and profile.html) when using [static exports](https://nextjs.org/docs/app/building-your-application/deploying/static-exports).
> 
> Our Django app structure is as follows:
> 
> ├── app/ │ └── templates/ │ └── index.html (created with yarn build in React) └── static/ ├── css ├── js └── img
> 
> I would appreciate your advice on how to proceed with this use case. Would there be any disadvantages if I choose to use "static exports"?
> 
> ### Additional information
> _No response_
> 
> ### Example
> _No response_

It probably it is better for you to serve your nextjs and your dango server to reverse proxy.
This way you can use all features of nextjs
